src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp

```diff
@@ -493,7 +518,10 @@ void PrettyBlockOutputFormat::writeSuffix()
 
     if (total_rows >= format_settings.pretty.max_rows)
```

Comment:
When `output_format_pretty_max_rows` equal to `LIMIT` it writes "Showed first":
```
SELECT *
FROM system.numbers
LIMIT 3
FORMAT PrettyMonoBlock
SETTINGS output_format_pretty_max_rows = 3

Query id: 7f513247-acf0-4f01-95d0-de705b1a90ea

   ┏━━━━━━━━┓
   ┃ number ┃
   ┡━━━━━━━━┩
1. │      0 │
   ├────────┤
2. │      1 │
   ├────────┤
3. │      2 │
   └────────┘
  Showed first 3.

3 rows in set. Elapsed: 0.002 sec.
```

It seems the condition should be `total_rows > format_settings.pretty.max_rows`